posthog/src/main/java/com/posthog/PostHogInterface.kt

```diff
@@ -0,0 +1,138 @@
+package com.posthog
+
+/**
+ * The PostHog SDK entry point
+ */
+public interface PostHogInterface {
+
+    /**
+     * Setup the SDK
+     * @param config the Config class
+     */
+    public fun <T : PostHogConfig> setup(config: T)
+
+    /**
+     * Closes the SDK
+     */
+    public fun close()
+
+    /**
+     * Captures events
+     * @param distinctId the distinctId, the generated distinctId is used if not given
+     * @param properties the custom properties
+     * @param userProperties the user properties, set as a "$set" property
+     * @param userPropertiesSetOnce the user properties to set only once, set as a "$set_once" property
+     * @param groupProperties the group properties, set as a "$groups" property
+     */
+    public fun capture(
+        event: String,
+        distinctId: String? = null,
+        properties: Map<String, Any>? = null,
+        userProperties: Map<String, Any>? = null,
+        userPropertiesSetOnce: Map<String, Any>? = null,
+        groupProperties: Map<String, Any>? = null,
+    )
+
+    /**
+     * Identifies the user
+     * @param distinctId the distinctId
+     * @param properties the custom properties
+     * @param userProperties the user properties, set as a "$set" property
+     * @param userPropertiesSetOnce the user properties to set only once, set as a "$set_once" property
+     */
+    public fun identify(
+        distinctId: String,
+        properties: Map<String, Any>? = null,
+        userProperties: Map<String, Any>? = null,
+        userPropertiesSetOnce: Map<String, Any>? = null,
+        // TODO: should we have groupProperties here?
+    )
+
+    /**
+     * Reloads the feature flags
+     * @param onFeatureFlags the callback to get notified once feature flags is ready to use
+     */
+    public fun reloadFeatureFlags(onFeatureFlags: PostHogOnFeatureFlags? = null)
```

Comment:
The JS SDK interface has a nice feature to register callbacks that are called when feature flags change. https://github.com/PostHog/posthog-js/blob/master/src/posthog-core.ts#L1172

Might be a stretch to include it now but would be nice to have...